src/proto/beats.proto

```diff
@@ -11,17 +11,36 @@ enum Source {
   USER = 2;
 }
 
-enum Type {
-  BEAT = 0;
-  BAR = 1;
-  PHRASE = 2;
+enum SectionType {
+  INTRO = 0;
+  INNER = 1;
+  OUTRO = 2;
 }
 
 message TimeSignature {
   optional int32 beats_per_bar = 1 [ default = 4];
   optional int32 note_value = 2 [ default = 4 ];
 }
 
+message TimeSignatureMarker {
+  optional int32 beat_index = 1 [ default = 0 ];
+  optional TimeSignature time_signature = 2;
+}
+
+message PhraseMarker {
+  optional int32 beat_index = 1 [ default = 0 ];
+}
+
+message Section {
+  optional string name = 1 [ default = "UNNAMED"];
+  optional SectionType type = 2 [ default = INNER ];
```

Comment:
I think sections should be able to have a color. What type can we use to store an RGB color in a protobuf?

Reply:
We can define a new message color
```
int32 red
int32 green
int32 blue
double alpha
```

Reply:
I do not think we need an alpha channel for section colors. We explicitly chose to not have an alpha channel for cue and track colors.

Reply:
Oh true. I remember that :/
We'll only use solid colors on the waveform.